### PR TITLE
feat(apigateway): support for API Key values and names

### LINKED ISSUE #3233: API Gateway: Custom API key value and API key export
* **I'm submitting a ...**
  - [ ] :beetle: bug report
  - [x] :rocket: feature request
  - [x] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

Please add possibility to add custom value and modification for API key in API Gateway.
This feature supported by CFN - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-apikey.html#cfn-apigateway-apikey-value

Also there is need to be able to import existing API keys. For example to be able to migrate to CDK from CFN or manually created resources. And cross-stack interactions.

### LINKED ISSUE #7767: Custom names for API keys in API Gateway
Currently you can only create API keys with `api_key = api.add_api_key("foobar")` but the name/ID given in that call is not used. This is really inconvenient if you for example want to create multiple API keys for different users. You should be able to name them `apikey-customer1` etc, instead they are now all called `my-stack-2934209384`.

### Use Case

I have multiple systems calling my API and I want to have separate usage plans and API keys for them with sensible names. Sensible names help in managing the keys.

### Proposed Solution

Most constructs take an optional `name` parameter which then overrides the CFN auto-generated name-hash. Same should be possible for API keys.

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by njlynch**
### Commit Message
feat(apigateway): specify API key name and value in `addApiKey()` (#7714)

fixes #3233
fixes #7767
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-apigateway/lib/restapi.ts [hunk 1]
```diff
@@ -328,11 +328,12 @@ export class RestApi extends Resource implements IRestApi {
   }
 
   /**
-   * Add an ApiKey
+   * Add an ApiKey, optionally with a supplied key value. If provided, the key value must be at least 20 characters.
    */
-  public addApiKey(id: string): IApiKey {
+  public addApiKey(id: string, value?: string): IApiKey {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/restapi.ts:333]: This change will make the API less extendible. 

Instead create a new interface called `ApiKeyOptions` which will be inherited into `ApiKeyProps`. The second parameter of this API call will be of the type `ApiKeyOptions`.

Here's an example - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L666-L682

Comment (nija-at) [reply packages/@aws-cdk/aws-apigateway/lib/restapi.ts:333]: By doing this, and moving `apiKeyName` to `ApiKeyOptions`, you will be able to fix this as well - https://github.com/aws/aws-cdk/issues/7767 😊

File: packages/@aws-cdk/aws-apigateway/README.md [hunk 2]
```diff
@@ -177,6 +177,13 @@ plan.addApiStage({
 });
 ```
 
+The value of the API Key will be automatically generated by API Gateway.
+Alternatively, the API Key value can be specified during key creation.
+
+```ts
+const key = api.addApiKey('ApiKey', 'MyApiKeyThatIsAtLeast20Characters');
+```
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/README.md:185]: Documentation is not aligned with code.

Comment (njlynch) [reply packages/@aws-cdk/aws-apigateway/README.md]: D'oh. Whoops! Posted an update to the README.

**Pull request merged**